File: C++/Programming Basics with C++/09. Exercise - While-loop/simple-tasks/Walking.cpp
#include<iostream>
#include <string>

using namespace std;

int main() {
    int sum = 0;
    while (sum < 10000) {
        string command;
        getline(cin, command);
        if (command == "Going home") {
            getline(cin, command);
            sum += stoi(command);
            break;
        }
        sum += stoi(command);
    }
    if (sum >= 10000) {
        cout << "Goal reached! Good job!" << endl;
    } else {
        cout << 10000 - sum << " more steps to reach goal." << endl;
    }
    return 0;
}

File: C#/Programming Basics with C#/08. While-Loop/05. Walking/Walking.cs
using System;

class Walking
{
    static void Main()
    {
        var a = "a";
        int sum = 0;
        while (a != "Going home" && sum < 10000)
        {
            a = Console.ReadLine();

            if (a == "Going home")
            {
                sum = sum + int.Parse(Console.ReadLine());
            }
            else
            {
                sum = sum + int.Parse(a);
            }
        }
        if (sum >= 10000)
        {
            Console.WriteLine("Goal reached! Good job!");
        }
        else
        {
            Console.WriteLine($"{10000 - sum} more steps to reach goal.");
        }
    }
}
